src/main/java/javaslang/collection/Iterator.java

```diff
@@ -128,6 +128,82 @@ public T next() {
     }
 
     /**
+     * Creates an Iterator which traverses along all given iterators.
+     * @param iterators The list of iterators
+     * @param <T>       Component type.
+     * @return A new {@code javaslang.collection.Iterator}
+     */
+    @SafeVarargs
+    @SuppressWarnings({"unchecked", "varargs"})
+    static <T> Iterator<T> ofIterators(Iterator<? extends T>... iterators) {
+        Objects.requireNonNull(iterators, "iterators is null");
+        return iterators.length == 0 ? Iterator.empty() : new ConcatIterator<>(Stream.of(iterators).iterator());
+    }
+
+    /**
+     * Creates an Iterator which traverses along all given iterables.
+     * @param iterables The list of iterables
+     * @param <T>       Component type.
+     * @return A new {@code javaslang.collection.Iterator}
+     */
+    @SafeVarargs
+    @SuppressWarnings({"unchecked", "varargs"})
+    static <T> Iterator<T> ofIterables(Iterable<? extends T>... iterables) {
+        Objects.requireNonNull(iterables, "iterables is null");
+        return iterables.length == 0 ? Iterator.empty() : new ConcatIterator<>(Stream.of(iterables).map(Iterator::ofAll).iterator());
+    }
+
+    /**
+     * Creates an Iterator which traverses along all given iterators.
+     * @param iterators The iterator over iterators
+     * @param <T>       Component type.
+     * @return A new {@code javaslang.collection.Iterator}
+     */
+    static <T> Iterator<T> ofIterators(Iterator<Iterator<T>> iterators) {
```

Comment:
The input parameter can be slightly more general, e.g. `Iterator<? extends Iterator<T>>` if not even Iterator<? extends Iterator<? extends T>>` which would allow to concat two iterators `Iterator<Integer>` and `ConcatIterator<Double>` to an iterator of Number.